torchaudio/backend/sox_io_backend.py

```diff
@@ -16,17 +16,10 @@ def _info(
         filepath: str,
         format: Optional[str] = None,
 ) -> AudioMetaData:
-    if hasattr(filepath, 'read'):
-        sinfo = torchaudio._torchaudio.get_info_fileobj(filepath, format)
-        return AudioMetaData(*sinfo)
-    sinfo = torch.ops.torchaudio.sox_io_get_info(os.fspath(filepath), format)
-    return AudioMetaData(
-        sinfo.get_sample_rate(),
-        sinfo.get_num_frames(),
-        sinfo.get_num_channels(),
-        sinfo.get_bits_per_sample(),
-        sinfo.get_encoding(),
-    )
+    sinfo = (torchaudio._torchaudio.get_info_fileobj(filepath, format)
+             if hasattr(filepath, 'read') else
+             torch.ops.torchaudio.sox_io_get_info(os.fspath(filepath), format))
+    return AudioMetaData(*sinfo)
```

Comment:
Can you try merge `_info` func and `info` in the same way as `load` function?

I think now TorchScript compiler can handle it without extracting the Python-only part to a helper function.

You can run the test locally with `pytest test/torchaudio_unittest/sox_io_backend/torchscript_test.py`